substrate/frame/staking/src/ledger.rs

```diff
@@ -166,13 +170,23 @@ impl<T: Config> StakingLedger<T> {
 			return Err(Error::<T>::NotStash)
 		}
 
+		let controller = &self.controller().ok_or_else(|| {
+			defensive!("update called on a ledger that was not contructed via `StakingLedger` impl. unexpected.");
+			Error::<T>::NotController
+        })?;
+
+		// previous stake is the current stake in storage.
+		let prev_stake: Option<Stake<_>> = match Ledger::<T>::get(&controller) {
+			Some(current_ledger) => Some(current_ledger.into()),
+			None => None,
+		};
+
 		T::Currency::set_lock(STAKING_ID, &self.stash, self.total, WithdrawReasons::all());
-		Ledger::<T>::insert(
-			&self.controller().ok_or_else(|| {
-				defensive!("update called on a ledger that is not bonded.");
-				Error::<T>::NotController
-			})?,
-			&self,
+		Ledger::<T>::insert(controller, &self);
+
+		<T::EventListeners as OnStakingUpdate<T::AccountId, BalanceOf<T>>>::on_stake_update(
```

Comment:
Should be fine simply like this?
```suggestion
		T::EventListeners:on_stake_update(
```

Reply:
Fixed!